c++/src/kj/async.h

```diff
@@ -388,6 +393,12 @@ class ForkedPromise {
   Promise<T> addBranch();
   // Add a new branch to the fork.  The branch is equivalent to the original promise.
 
+  _::BranchCoroPromise<T> addCoroBranch();
```

Comment:
Maybe `addBranchForCoAwait()`? The important thing being communicated here is: the caller is expected to immediately co_await the result.